src/server/rest/v1/service/AssetService.ts

```diff
@@ -458,24 +385,11 @@ export default class AssetService {
     // Check if component is active
     UtilsService.assertComponentIsActiveFromToken(req.user, TenantComponents.ASSET,
       Action.LIST, Entity.ASSET, MODULE_NAME, 'handleGetAssets');
-    // Check auth
-    if (!await Authorizations.canListAssets(req.user)) {
-      throw new AppAuthError({
-        errorCode: HTTPAuthError.FORBIDDEN,
-        user: req.user,
-        action: Action.LIST, entity: Entity.ASSET,
-        module: MODULE_NAME, method: 'handleGetAssets'
-      });
-    }
     // Filter
     const filteredRequest = AssetValidator.getInstance().validateAssetsGetReq(req.query);
-    // Get authorization filters
-    const authorizationAssetsFilters = await AuthorizationService.checkAndGetAssetsAuthorizations(
-      req.tenant, req.user, filteredRequest);
-    if (!authorizationAssetsFilters.authorized) {
-      UtilsService.sendEmptyDataResult(res, next);
-      return;
-    }
+    // Check dynamic auth
+    const authorizationAssetsFilter = await AuthorizationService.checkAndGetAssetsAuthorizations(
+      req.tenant, req.user, Action.LIST, filteredRequest);
```

Comment:
I would expect this code just after (cf. SiteService endpoint):
```
    if (! authorizationAssetsFilter.authorized) {
      UtilsService.sendEmptyDataResult(res, next);
      return;
    }
```

Reply:
checkAndGetAssetsAuthorizations throws an exception if not auhorized

Reply:
Done as discussed